torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1898,7 +1898,7 @@ void ProcessGroupNCCL::workEnqueue(
 }
 
 ProcessGroupNCCL::Options::Options(bool is_high_priority_stream)
-    : Backend::Options(NCCL_BACKEND_NAME, kProcessGroupNCCLDefaultTimeout),
+    : Backend::Options(NCCL_BACKEND_NAME),
```

Comment:
Okay I think I understand the confusion now, we are basically never using this pybind constructor (maybe it is used in some tests where we instantiate ProcessGroupNCCL directly rather than through `init_pg`):

https://github.com/pytorch/pytorch/blob/a1ab22b81d74a26873243d766d78724b3b70d570/torch/csrc/distributed/c10d/init.cpp#L2246-L2255

Instead in `init_pg` we always pass in `options` into the constructor so we always go through this path:

https://github.com/pytorch/pytorch/blob/a1ab22b81d74a26873243d766d78724b3b70d570/torch/csrc/distributed/c10d/init.cpp#L2239-L2245

So basically this change will change back to the original behavior of 30mins timeout and wont use `kProcessGroupNCCLDefaultTimeout`. But we need this change and the change in the stack above for the desired behavior, so im stamping it